api/controllers/UserController.js

```diff
@@ -214,26 +215,168 @@ module.exports = {
+    }
+    try {
+      var duration = req.params.duration ? parseInt(req.params.duration) : 0;
+      if (duration < 0) {
+        return res.json({error: "Invalid duration"}, 400);
       }
-
-      user.banned = req.allParams().ban;
-      user.save(function (err) {
+    } catch (err) {
+      return res.json({error: "Invalid duration"}, 400);
+    }
+    try {
+      for (var FC = 0; FC < req.params.additionalFCs.length; FC++) {
+        if (!req.params.additionalFCs[FC].match(/^(\d{4}-){2}\d{4}$/g)) {
+          return res.json({error: "Invalid friendcode list"}, 400);
+        }
+      }
+    }
+    catch (invalidfcerr) {
+      return res.json({error: "Invalid friendcode list"}, 400);
+    }
+    User.findOne({name: req.params.username}, function (finding_user_error, user) {
+      Reddit.getFlair(req.user.redToken, function (err, flair1, flair2) {
         if (err) {
           return res.json(err, 500);
         }
-        res.json(user, 200);
-      });
+        if (flair1) {
+          if (flair1.flair_css_class.indexOf(' ') === -1) {
+            flair1.flair_css_class += ' banned';
+          } else {
+            flair1.flair_css_class = flair1.flair_css_class.substring(0, flair1.flair_css_class.indexOf(' ')) + ' banned';
+          }
+        } else {
+          flair1 = {flair_css_class: 'default banned'};
+          flair1.flair_text = '';
+        }
+        if (flair2) {
+          if (flair2.flair_css_class) {
+            flair2.flair_css_class+=' banned';
+          }
+          else {
+            flair2.flair_css_class = 'banned';
+          }
+        } else {
+          flair2 = {flair_css_class: 'banned'};
+        }
+        var logged_fcs;
+        if (user) {
+          logged_fcs = user.loggedFriendCodes;
+        }
+        var unique_fcs = _.union(
+          flair1.flair_text.match(/(\d{4}-){2}\d{4}/g),
+          flair2.flair_text.match(/(\d{4}-){2}\d{4}/g),
+          logged_fcs,
+          req.params.additionalFCs
+        );
+        var igns = flair1.flair_text.substring(flair1.flair_text.indexOf("||") + 3);
+        var ptradesBanPromise = new Promise(function(resolve, reject) {
+          Ban.banFromSub(req.user.redToken, req.params.username, req.params.banMessage, req.params.banNote, 'pokemontrades', duration, resolve, reject);
+        });
+        var svexBanPromise = new Promise(function(resolve, reject) {
+          Ban.banFromSub(req.user.redToken, req.params.username, req.params.banMessage, req.params.banNote, 'SVExchange', duration, resolve, reject);
+        });
+        var promises;
+        if (duration) {
+          promises = [ //Tasks for tempbanning
+            ptradesBanPromise, 
+            svexBanPromise
+          ];
+        } else {
+          var ptradesFlairPromise = new Promise(function(resolve, reject) {
+            Ban.giveBannedUserFlair(req.user.redToken, req.params.username, flair1.flair_css_class, flair1.flair_text, 'pokemontrades', resolve, reject);
```

Comment:
Last thing (I promise).

Don't make the promises here. do it within ban itself, this is just wasted characters, and there's no need to make the promise here, then pass the res/rej into the inner function. Just make it inside the Ban function itself, and return the promise you make. Much nicer.